obal/data/roles/diff_brew_package/tasks/dist.yml

```diff
@@ -0,0 +1,44 @@
+- name: "Set branch checks"
+  set_fact:
+    "{{ item.name }}_branch_exists": "{{ true if item.releaser in releasers else false }}"
+  tags:
+    - always
+
+- name: 'Look up current version of package in {{ build_package_koji_command }} for {{ item.dist }}'
+  shell: "{{ build_package_koji_command }} list-tagged --quiet \
+    {{ item.koji_tag }} \
+    {{ inventory_hostname }} \
+    | sort --reverse --key 1 --version-sort \
+    | head -n1 \
+    | cut -d' ' -f1"
+  register: "latest_brewed_{{ item.name }}"
+  changed_when: False
+  when: "hostvars[inventory_hostname][item.name + '_branch_exists'] | bool"
+  tags:
+    - always
+
+- name: 'Calculate {{ item.dist }} RPM version in local specfile'
+  command: "rpmspec --query '{{ spec_file_path }}' \
+    --define 'dist {{ item.dist }}' \
+    --define 'foremandist .{{ foremandist }}' \
```

Comment:
This is rather specific to Foreman. Maybe a more generic approach is needed.

Reply:
Yup, sure is.  Still working on tackling it all